
Property changes on: .
___________________________________________________________________
Modified: svn:ignore
   - build
dist
*~
velocity.log
build.properties
.idea
*.iml
*.ipr
*.iws
.project
.classpath
.settings
prj.el
bin
bin.*
pom.xml

   + build
dist
*~
velocity.log
build.properties
.idea
*.iml
*.ipr
*.iws
.project
.classpath
.settings
prj.el
bin
bin.*
maven-build



Property changes on: solr
___________________________________________________________________
Modified: svn:ignore
   - build
*.iml
pom.xml
package
dist
.project
.classpath
.settings
bin
luke
build.properties

   + build
*.iml
package
dist
.project
.classpath
.settings
bin
luke
build.properties



Property changes on: solr/contrib
___________________________________________________________________
Deleted: svn:ignore
   - 
pom.xml



Property changes on: solr/contrib/clustering
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: solr/contrib/extraction
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: solr/contrib/dataimporthandler
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: solr/contrib/dataimporthandler-extras
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: solr/contrib/analysis-extras
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: solr/contrib/uima
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: solr/contrib/langid
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: solr/contrib/velocity
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: solr/test-framework
___________________________________________________________________
Deleted: svn:ignore
   - pom.xml



Property changes on: solr/webapp
___________________________________________________________________
Deleted: svn:ignore
   - pom.xml



Property changes on: solr/core
___________________________________________________________________
Deleted: svn:ignore
   - pom.xml



Property changes on: solr/solrj
___________________________________________________________________
Deleted: svn:ignore
   - pom.xml



Property changes on: lucene
___________________________________________________________________
Modified: svn:ignore
   - build
dist
*~
velocity.log
build.properties
.idea
*.iml
*.ipr
*.iws
.project
.classpath
.settings
prj.el
pom.xml
ivy

   + build
dist
*~
velocity.log
build.properties
.idea
*.iml
*.ipr
*.iws
.project
.classpath
.settings
prj.el
ivy



Property changes on: lucene/memory
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/test-framework
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/queries
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/suggest
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/facet
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/queryparser
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/analysis
___________________________________________________________________
Deleted: svn:ignore
   - pom.xml



Property changes on: lucene/analysis/kuromoji
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/analysis/icu
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/analysis/smartcn
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/analysis/common
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml




Property changes on: lucene/analysis/morfologik
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/analysis/phonetic
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/analysis/stempel
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/analysis/uima
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/demo
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/core
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/benchmark
___________________________________________________________________
Modified: svn:ignore
   - temp
work
*.iml
pom.xml

   + temp
work
*.iml



Property changes on: lucene/grouping
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/misc
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/spatial
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/highlighter
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml



Property changes on: lucene/sandbox
___________________________________________________________________
Modified: svn:ignore
   - pom.xml
*.iml

   + *.iml



Property changes on: lucene/join
___________________________________________________________________
Modified: svn:ignore
   - *.iml
pom.xml

   + *.iml


Index: dev-tools/maven/solr/webapp/pom.xml.template
===================================================================
--- dev-tools/maven/solr/webapp/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/solr/webapp/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Apache Solr Search Server</description>
   <properties>
     <module-directory>solr/webapp</module-directory>
-    <build-directory>../build</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -101,8 +102,8 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-war-plugin</artifactId>
         <configuration>
-          <warSourceDirectory>web</warSourceDirectory>
-          <webXml>web/WEB-INF/web.xml</webXml>
+          <warSourceDirectory>${module-path}/web</warSourceDirectory>
+          <webXml>${module-path}/web/WEB-INF/web.xml</webXml>
         </configuration>
       </plugin>
       <plugin>
@@ -114,11 +115,11 @@
           <webAppConfig>
             <contextPath>/solr</contextPath>
           </webAppConfig>
-          <webAppSourceDirectory>web</webAppSourceDirectory>
+          <webAppSourceDirectory>${module-path}/web</webAppSourceDirectory>
           <systemProperties>
             <systemProperty>
               <name>solr.solr.home</name>
-              <value>../example/solr</value>
+              <value>${top-level}/solr/example/solr</value>
             </systemProperty>
           </systemProperties>
         </configuration>
Index: dev-tools/maven/solr/contrib/pom.xml.template
===================================================================
--- dev-tools/maven/solr/contrib/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/solr/contrib/pom.xml.template	(working copy)
@@ -41,7 +41,6 @@
     <module>velocity</module>
   </modules>
   <build>
-    <directory>../build/solr-contrib-aggregator</directory>
     <plugins>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
Index: dev-tools/maven/solr/contrib/dataimporthandler-extras/pom.xml.template
===================================================================
--- dev-tools/maven/solr/contrib/dataimporthandler-extras/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/solr/contrib/dataimporthandler-extras/pom.xml.template	(working copy)
@@ -33,7 +33,9 @@
   <description>Apache Solr DataImportHandler Extras</description>
   <properties>
     <module-directory>solr/contrib/dataimporthandler-extras</module-directory>
-    <build-directory>../../build/contrib/solr-dataimporthandler-extras</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
+    <surefire-top-level>${top-level}/../..</surefire-top-level>
   </properties>
   <scm>
     <connection>
@@ -93,17 +95,14 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
-        <directory>src/test-files</directory>
+        <directory>${module-path}/src/test-files</directory>
       </testResource>
       <testResource>
-        <directory>../../core/src/test-files</directory>
+        <directory>${top-level}/solr/core/src/test-files</directory>
       </testResource>
     </testResources>
     <plugins>
@@ -112,7 +111,7 @@
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
           <systemPropertyVariables>
-            <java.util.logging.config.file>../../../../testlogging.properties</java.util.logging.config.file>
+            <java.util.logging.config.file>${surefire-top-level}/solr/testlogging.properties</java.util.logging.config.file>
           </systemPropertyVariables>
         </configuration>
       </plugin>
Index: dev-tools/maven/solr/contrib/analysis-extras/pom.xml.template
===================================================================
--- dev-tools/maven/solr/contrib/analysis-extras/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/solr/contrib/analysis-extras/pom.xml.template	(working copy)
@@ -33,7 +33,9 @@
   <description>Apache Solr Analysis Extras</description>
   <properties>
     <module-directory>solr/contrib/analysis-extras</module-directory>
-    <build-directory>../../build/contrib/solr-analysis-extras</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
+    <surefire-top-level>${top-level}/../..</surefire-top-level>
   </properties>
   <scm>
     <connection>
@@ -90,17 +92,14 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
-        <directory>src/test-files</directory>
+        <directory>${module-path}/src/test-files</directory>
       </testResource>
       <testResource>
-        <directory>../../core/src/test-files</directory>
+        <directory>${top-level}/solr/core/src/test-files</directory>
       </testResource>
     </testResources>
     <plugins>
@@ -109,7 +108,7 @@
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
           <systemPropertyVariables>
-            <java.util.logging.config.file>../../../../testlogging.properties</java.util.logging.config.file>
+            <java.util.logging.config.file>${surefire-top-level}/solr/testlogging.properties</java.util.logging.config.file>
           </systemPropertyVariables>
         </configuration>
       </plugin>
Index: dev-tools/maven/solr/contrib/uima/pom.xml.template
===================================================================
--- dev-tools/maven/solr/contrib/uima/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/solr/contrib/uima/pom.xml.template	(working copy)
@@ -33,7 +33,9 @@
   <description>Apache Solr - UIMA integration</description>
   <properties>
     <module-directory>solr/contrib/uima</module-directory>
-    <build-directory>../../build/contrib/solr-uima</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
+    <surefire-top-level>${top-level}/../..</surefire-top-level>
   </properties>
   <scm>
     <connection>
@@ -108,19 +110,16 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <resources>
       <resource>
-        <directory>src/resources</directory>
+        <directory>${module-path}/src/resources</directory>
       </resource>
     </resources>
     <testResources>
       <testResource>
-        <directory>src/test-files</directory>
+        <directory>${module-path}/src/test-files</directory>
       </testResource>
     </testResources>
     <plugins>
@@ -129,7 +128,7 @@
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
           <systemPropertyVariables>
-            <java.util.logging.config.file>../../../../testlogging.properties</java.util.logging.config.file>
+            <java.util.logging.config.file>${surefire-top-level}/solr/testlogging.properties</java.util.logging.config.file>
           </systemPropertyVariables>
         </configuration>
       </plugin>
Index: dev-tools/maven/solr/contrib/langid/pom.xml.template
===================================================================
--- dev-tools/maven/solr/contrib/langid/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/solr/contrib/langid/pom.xml.template	(working copy)
@@ -37,7 +37,9 @@
   </description>
   <properties>
     <module-directory>solr/contrib/langid</module-directory>
-    <build-directory>../../build/contrib/solr-langid</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
+    <surefire-top-level>${top-level}/../..</surefire-top-level>
   </properties>
   <scm>
     <connection>
@@ -91,22 +93,19 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <resources>
       <resource>
-        <directory>src/resources</directory>
+        <directory>${module-path}/src/resources</directory>
       </resource>
     </resources>
     <testResources>
       <testResource>
-        <directory>src/test-files</directory>
+        <directory>${module-path}/src/test-files</directory>
       </testResource>
       <testResource>
-        <directory>../../core/src/test-files</directory>
+        <directory>${top-level}/solr/core/src/test-files</directory>
       </testResource>
     </testResources>
     <plugins>
@@ -115,7 +114,7 @@
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
           <systemPropertyVariables>
-            <java.util.logging.config.file>../../../../testlogging.properties</java.util.logging.config.file>
+            <java.util.logging.config.file>${surefire-top-level}/solr/testlogging.properties</java.util.logging.config.file>
           </systemPropertyVariables>
         </configuration>
       </plugin>
Index: dev-tools/maven/solr/contrib/velocity/pom.xml.template
===================================================================
--- dev-tools/maven/solr/contrib/velocity/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/solr/contrib/velocity/pom.xml.template	(working copy)
@@ -33,7 +33,9 @@
   <description>Apache Solr Velocity</description>
   <properties>
     <module-directory>solr/contrib/velocity</module-directory>
-    <build-directory>../../build/contrib/solr-velocity</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
+    <surefire-top-level>${top-level}/../..</surefire-top-level>
   </properties>
   <scm>
     <connection>
@@ -131,17 +133,14 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
-        <directory>src/test-files</directory>
+        <directory>${module-path}/src/test-files</directory>
       </testResource>
       <testResource>
-        <directory>../../core/src/test-files</directory>
+        <directory>${top-level}/solr/core/src/test-files</directory>
       </testResource>
     </testResources>
     <plugins>
@@ -150,7 +149,7 @@
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
           <systemPropertyVariables>
-            <java.util.logging.config.file>../../../../testlogging.properties</java.util.logging.config.file>
+            <java.util.logging.config.file>${surefire-top-level}/solr/testlogging.properties</java.util.logging.config.file>
           </systemPropertyVariables>
         </configuration>
       </plugin>
Index: dev-tools/maven/solr/contrib/clustering/pom.xml.template
===================================================================
--- dev-tools/maven/solr/contrib/clustering/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/solr/contrib/clustering/pom.xml.template	(working copy)
@@ -33,7 +33,9 @@
   <description>Apache Solr Clustering</description>
   <properties>
     <module-directory>solr/contrib/clustering</module-directory>
-    <build-directory>../../build/contrib/solr-clustering</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
+    <surefire-top-level>${top-level}/../..</surefire-top-level>
   </properties>
   <scm>
     <connection>
@@ -95,17 +97,14 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
-        <directory>src/test-files</directory>
+        <directory>${module-path}/src/test-files</directory>
       </testResource>
       <testResource>
-        <directory>../../core/src/test-files</directory>
+        <directory>${top-level}/solr/core/src/test-files</directory>
       </testResource>
     </testResources>
     <plugins>
@@ -114,7 +113,7 @@
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
           <systemPropertyVariables>
-            <java.util.logging.config.file>../../../../testlogging.properties</java.util.logging.config.file>
+            <java.util.logging.config.file>${surefire-top-level}/solr/testlogging.properties</java.util.logging.config.file>
           </systemPropertyVariables>
         </configuration>
       </plugin>
Index: dev-tools/maven/solr/contrib/extraction/pom.xml.template
===================================================================
--- dev-tools/maven/solr/contrib/extraction/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/solr/contrib/extraction/pom.xml.template	(working copy)
@@ -36,7 +36,9 @@
   </description>
   <properties>
     <module-directory>solr/contrib/extraction</module-directory>
-    <build-directory>../../build/contrib/solr-cell</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
+    <surefire-top-level>${top-level}/../..</surefire-top-level>
   </properties>
   <scm>
     <connection>
@@ -91,17 +93,14 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
-        <directory>src/test-files</directory>
+        <directory>${module-path}/src/test-files</directory>
       </testResource>
       <testResource>
-        <directory>../../core/src/test-files</directory>
+        <directory>${top-level}/solr/core/src/test-files</directory>
       </testResource>
     </testResources>
     <plugins>
@@ -110,7 +109,7 @@
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
           <systemPropertyVariables>
-            <java.util.logging.config.file>../../../../testlogging.properties</java.util.logging.config.file>
+            <java.util.logging.config.file>${surefire-top-level}/solr/testlogging.properties</java.util.logging.config.file>
           </systemPropertyVariables>
         </configuration>
       </plugin>
Index: dev-tools/maven/solr/contrib/dataimporthandler/pom.xml.template
===================================================================
--- dev-tools/maven/solr/contrib/dataimporthandler/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/solr/contrib/dataimporthandler/pom.xml.template	(working copy)
@@ -33,7 +33,9 @@
   <description>Apache Solr DataImportHandler</description>
   <properties>
     <module-directory>solr/contrib/dataimporthandler</module-directory>
-    <build-directory>../../build/contrib/solr-dataimporthandler</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
+    <surefire-top-level>${top-level}/../..</surefire-top-level>
   </properties>
   <scm>
     <connection>
@@ -74,17 +76,14 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
-        <directory>src/test-files</directory>
+        <directory>${module-path}/src/test-files</directory>
       </testResource>
       <testResource>
-        <directory>../../core/src/test-files</directory>
+        <directory>${top-level}/solr/core/src/test-files</directory>
       </testResource>
     </testResources>
     <plugins>
@@ -104,7 +103,7 @@
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
           <systemPropertyVariables>
-            <java.util.logging.config.file>../../../../testlogging.properties</java.util.logging.config.file>
+            <java.util.logging.config.file>${surefire-top-level}/solr/testlogging.properties</java.util.logging.config.file>
           </systemPropertyVariables>
         </configuration>
       </plugin>
Index: dev-tools/maven/solr/pom.xml.template
===================================================================
--- dev-tools/maven/solr/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/solr/pom.xml.template	(working copy)
@@ -100,7 +100,6 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>build/solr-parent</directory>
     <pluginManagement>
       <plugins>
         <plugin>
Index: dev-tools/maven/solr/core/pom.xml.template
===================================================================
--- dev-tools/maven/solr/core/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/solr/core/pom.xml.template	(working copy)
@@ -33,7 +33,9 @@
   <description>Apache Solr Core</description>
   <properties>
     <module-directory>solr/core</module-directory>
-    <build-directory>../build/solr-maven</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
+    <surefire-top-level>${top-level}/../..</surefire-top-level>
   </properties>
   <scm>
     <connection>
@@ -222,14 +224,11 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
-        <directory>src/test-files</directory>
+        <directory>${module-path}/src/test-files</directory>
       </testResource>
       <testResource>
         <directory>${project.build.testSourceDirectory}</directory>
@@ -238,7 +237,7 @@
         </excludes>
       </testResource>
       <testResource>
-        <directory>../solrj/src/test-files</directory>
+        <directory>${top-level}/solr/solrj/src/test-files</directory>
       </testResource>
     </testResources>
     <plugins>
@@ -247,7 +246,7 @@
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
           <systemPropertyVariables>
-            <java.util.logging.config.file>../../../testlogging.properties</java.util.logging.config.file>
+            <java.util.logging.config.file>${surefire-top-level}/solr/testlogging.properties</java.util.logging.config.file>
           </systemPropertyVariables>
         </configuration>
       </plugin>
@@ -294,8 +293,8 @@
             </goals>
             <configuration>
               <sources>
-                <source>../test-framework/src/java</source>
-                <source>../solrj/src/test</source>  <!-- solrj tests are run from solr-core -->
+                <source>${top-level}/solr/test-framework/src/java</source>
+                <source>${top-level}/solr/solrj/src/test</source>  <!-- solrj tests are run from solr-core -->
               </sources>
             </configuration>
           </execution>
Index: dev-tools/maven/solr/solrj/pom.xml.template
===================================================================
--- dev-tools/maven/solr/solrj/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/solr/solrj/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Apache Solr Solrj</description>
   <properties>
     <module-directory>solr/solrj</module-directory>
-    <build-directory>../build/solr-solrj</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -122,11 +123,8 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory/>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources/>
     <plugins>
       <plugin>
Index: dev-tools/maven/solr/test-framework/pom.xml.template
===================================================================
--- dev-tools/maven/solr/test-framework/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/solr/test-framework/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Apache Solr Test Framework</description>
   <properties>
     <module-directory>solr/test-framework</module-directory>
-    <build-directory>../build/solr-test-framework</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -82,14 +83,11 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory/>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <resources>
       <resource>
-        <directory>.</directory>
+        <directory>${module-path}</directory>
         <excludes>
           <exclude>**/*.java</exclude>
         </excludes>
Index: dev-tools/maven/README.maven
===================================================================
--- dev-tools/maven/README.maven	(revision 1334608)
+++ dev-tools/maven/README.maven	(working copy)
@@ -75,6 +75,7 @@
 
          svn update
          ant get-maven-poms
+         cd maven-build
 
    The details, followed by some example Maven commands:
 
@@ -84,9 +85,9 @@
       from the Apache Subversion repository, run "svn update" from the top
       level.
 
-   3. Copy the Maven POM templates from under dev-tools/maven/ to where they
-      they need to go in order to drive the Maven build, using the following
-      command from the top-level directory:
+   3. Copy the Maven POM templates from under dev-tools/maven/ to the
+      maven-build/ directory using the following command from the top-level
+      directory:
 
          ant get-maven-poms
 
@@ -102,7 +103,12 @@
 
          ant -Dversion=4.0-my-special-version get-maven-poms
 
+   4. To remove the maven-build/ directory and its contents, use the following
+      command from the top-level directory:
 
+         and clean-maven-build
+
+
    Some example Maven commands you can use after you perform the above
    preparatory steps:
 
Index: dev-tools/maven/pom.xml.template
===================================================================
--- dev-tools/maven/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/pom.xml.template	(working copy)
@@ -30,14 +30,15 @@
   <artifactId>lucene-solr-grandparent</artifactId>
   <version>@version@</version>
   <packaging>pom</packaging>
-  <name>Grandparent POM for Apache Lucene Java and Apache Solr</name>
-  <description>Parent POM for Apache Lucene Java and Apache Solr</description>
+  <name>Grandparent POM for Apache Lucene Core and Apache Solr</name>
+  <description>Parent POM for Apache Lucene Core and Apache Solr</description>
   <url>http://lucene.apache.org/java</url>
   <modules>
     <module>lucene</module>
     <module>solr</module>
   </modules>
   <properties>
+    <top-level>..</top-level>
     <base.specification.version>4.0.0</base.specification.version>
     <maven.build.timestamp.format>yyyy-MM-dd HH:mm:ss</maven.build.timestamp.format>
     <java.compat.version>1.6</java.compat.version>
@@ -60,6 +61,7 @@
     <tests.locale>random</tests.locale>
     <tests.luceneMatchVersion>4.0</tests.luceneMatchVersion>
     <tests.multiplier>1</tests.multiplier>
+    <tests.nightly>false</tests.nightly>
     <tests.postingsformat>random</tests.postingsformat>
     <tests.timezone>random</tests.timezone>
     <tests.verbose>false</tests.verbose>
@@ -397,7 +399,6 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>lucene/build/lucene-parent</directory>
     <pluginManagement>
       <plugins>
         <plugin>
@@ -480,6 +481,7 @@
           <artifactId>maven-surefire-plugin</artifactId>
           <version>2.12</version>
           <configuration>
+            <runOrder>random</runOrder>
             <reportFormat>plain</reportFormat>
             <workingDirectory>${project.build.directory}/test</workingDirectory>
             <argLine>-Xmx512M</argLine>
@@ -602,6 +604,7 @@
           <getRevisionOnlyOnce>true</getRevisionOnlyOnce>
           <revisionOnScmFailure>NO-REVISION-AVAILABLE</revisionOnScmFailure>
           <buildNumberPropertyName>svn.revision</buildNumberPropertyName>
+          <scmDirectory>${top-level}</scmDirectory>
         </configuration>
       </plugin>
       <plugin>
Index: dev-tools/maven/lucene/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/pom.xml.template	(working copy)
@@ -63,7 +63,4 @@
     <module>spatial</module>
     <module>suggest</module>
   </modules>
-  <build>
-    <directory>build/lucene-parent</directory>
-  </build>
 </project>
Index: dev-tools/maven/lucene/memory/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/memory/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/memory/pom.xml.template	(working copy)
@@ -35,7 +35,8 @@
   </description>
   <properties>
     <module-directory>lucene/memory</module-directory>
-    <build-directory>../build/memory</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -69,11 +70,8 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
         <directory>${project.build.testSourceDirectory}</directory>
Index: dev-tools/maven/lucene/test-framework/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/test-framework/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/test-framework/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Apache Lucene Java Test Framework</description>
   <properties>
     <module-directory>lucene/test-framework</module-directory>
-    <build-directory>../build/test-framework</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -66,12 +67,10 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
     <resources>
       <resource>
-        <directory>src/resources</directory>
+        <directory>${module-path}/src/resources</directory>
       </resource>
       <resource>
         <directory>${project.build.sourceDirectory}</directory>
Index: dev-tools/maven/lucene/queries/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/queries/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/queries/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Lucene Queries Module</description>
   <properties>
     <module-directory>lucene/queries</module-directory>
-    <build-directory>../build/queries</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -61,11 +62,8 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
         <directory>${project.build.testSourceDirectory}</directory>
Index: dev-tools/maven/lucene/suggest/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/suggest/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/suggest/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Lucene Suggest Module</description>
   <properties>
     <module-directory>lucene/suggest</module-directory>
-    <build-directory>../build/suggest</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -66,11 +67,8 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
         <directory>${project.build.testSourceDirectory}</directory>
Index: dev-tools/maven/lucene/facet/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/facet/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/facet/pom.xml.template	(working copy)
@@ -35,7 +35,8 @@
   </description>
   <properties>
     <module-directory>lucene/facet</module-directory>
-    <build-directory>../build/facet</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -68,11 +69,8 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
         <directory>${project.build.testSourceDirectory}</directory>
@@ -94,7 +92,7 @@
             </goals>
             <configuration>
               <sources>
-                <source>src/examples</source>
+                <source>${module-path}/src/examples</source>
               </sources>
             </configuration>
           </execution>
Index: dev-tools/maven/lucene/queryparser/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/queryparser/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/queryparser/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Lucene QueryParsers module</description>
   <properties>
     <module-directory>lucene/queryparser</module-directory>
-    <build-directory>../build/queryparser</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -71,16 +72,13 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
     <resources>
       <resource>
-        <directory>src/resources</directory>
+        <directory>${module-path}/src/resources</directory>
       </resource>
     </resources>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
         <directory>${project.build.testSourceDirectory}</directory>
Index: dev-tools/maven/lucene/analysis/smartcn/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/analysis/smartcn/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/analysis/smartcn/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Smart Chinese Analyzer</description>
   <properties>
     <module-directory>lucene/analysis/smartcn</module-directory>
-    <build-directory>../../build/analysis/smartcn</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -66,14 +67,11 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <resources>
       <resource>
-        <directory>src/resources</directory>
+        <directory>${module-path}/src/resources</directory>
       </resource>
     </resources>
     <testResources>
Index: dev-tools/maven/lucene/analysis/common/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/analysis/common/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/analysis/common/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Additional Analyzers</description>
   <properties>
     <module-directory>lucene/analysis/common</module-directory>
-    <build-directory>../../build/analysis/common</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -61,14 +62,11 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <resources>
       <resource>
-        <directory>src/resources</directory>
+        <directory>${module-path}/src/resources</directory>
       </resource>
     </resources>
     <testResources>
Index: dev-tools/maven/lucene/analysis/morfologik/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/analysis/morfologik/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/analysis/morfologik/pom.xml.template	(working copy)
@@ -35,7 +35,8 @@
   </description>
   <properties>
     <module-directory>lucene/analysis/morfologik</module-directory>
-    <build-directory>../../build/analysis/morfologik</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -72,11 +73,8 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
         <directory>${project.build.testSourceDirectory}</directory>
Index: dev-tools/maven/lucene/analysis/phonetic/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/analysis/phonetic/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/analysis/phonetic/pom.xml.template	(working copy)
@@ -35,7 +35,8 @@
   </description>
   <properties>
     <module-directory>lucene/analysis/phonetic</module-directory>
-    <build-directory>../../build/analysis/phonetic</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -72,11 +73,8 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
         <directory>${project.build.testSourceDirectory}</directory>
Index: dev-tools/maven/lucene/analysis/stempel/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/analysis/stempel/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/analysis/stempel/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Stempel Analyzer</description>
   <properties>
     <module-directory>lucene/analysis/stempel</module-directory>
-    <build-directory>../../build/analysis/stempel</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -66,14 +67,11 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <resources>
       <resource>
-        <directory>src/resources</directory>
+        <directory>${module-path}/src/resources</directory>
       </resource>
     </resources>
     <testResources>
Index: dev-tools/maven/lucene/analysis/uima/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/analysis/uima/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/analysis/uima/pom.xml.template	(working copy)
@@ -37,7 +37,8 @@
   </description>
   <properties>
     <module-directory>lucene/analysis/uima</module-directory>
-    <build-directory>../../build/analysis/uima</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -82,19 +83,16 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <resources>
       <resource>
-        <directory>src/resources</directory>
+        <directory>${module-path}/src/resources</directory>
       </resource>
     </resources>
     <testResources>
       <testResource>
-        <directory>src/test-files</directory>
+        <directory>${module-path}/src/test-files</directory>
       </testResource>
     </testResources>
   </build>
Index: dev-tools/maven/lucene/analysis/kuromoji/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/analysis/kuromoji/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/analysis/kuromoji/pom.xml.template	(working copy)
@@ -35,7 +35,8 @@
   </description>
   <properties>
     <module-directory>lucene/analysis/kuromoji</module-directory>
-    <build-directory>../../build/analysis/kuromoji</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -68,14 +69,11 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <resources>
       <resource>
-        <directory>src/resources</directory>
+        <directory>${module-path}/src/resources</directory>
       </resource>
     </resources>
     <testResources>
Index: dev-tools/maven/lucene/analysis/icu/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/analysis/icu/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/analysis/icu/pom.xml.template	(working copy)
@@ -36,7 +36,8 @@
   </description>
   <properties>
     <module-directory>lucene/analysis/icu</module-directory>
-    <build-directory>../../build/analysis/icu</build-directory>
+    <top-level>../../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -73,14 +74,11 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <resources>
       <resource>
-        <directory>src/resources</directory>
+        <directory>${module-path}/src/resources</directory>
       </resource>
     </resources>
     <testResources>
Index: dev-tools/maven/lucene/demo/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/demo/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/demo/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>This is the demo for Apache Lucene Java</description>
   <properties>
     <module-directory>lucene/demo</module-directory>
-    <build-directory>../build/demo</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -76,11 +77,8 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
         <directory>${project.build.testSourceDirectory}</directory>
Index: dev-tools/maven/lucene/core/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/core/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/core/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Apache Lucene Java Core</description>
   <properties>
     <module-directory>lucene/core</module-directory>
-    <build-directory>../build/lucene-maven</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -69,22 +70,19 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
     <resources>
       <resource>
-        <directory>src/resources</directory>
+        <directory>${module-path}/src/resources</directory>
       </resource>
     </resources>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
-        <directory>../test-framework/src/resources</directory>
+        <directory>${top-level}/lucene/test-framework/src/resources</directory>
       </testResource>
       <testResource>
-        <directory>../test-framework/src/java</directory>
+        <directory>${top-level}/lucene/test-framework/src/java</directory>
         <excludes>
           <exclude>**/*.java</exclude>
         </excludes>
@@ -152,7 +150,7 @@
             </goals>
             <configuration>
               <sources>
-                <source>../test-framework/src/java</source>
+                <source>${top-level}/lucene/test-framework/src/java</source>
               </sources>
             </configuration>
           </execution>
Index: dev-tools/maven/lucene/benchmark/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/benchmark/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/benchmark/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Lucene Benchmarking Module</description>
   <properties>
     <module-directory>lucene/benchmark</module-directory>
-    <build-directory>../build/benchmark</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -98,11 +99,8 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
         <directory>${project.build.testSourceDirectory}</directory>
@@ -111,7 +109,7 @@
         </excludes>
       </testResource>
       <testResource>
-        <directory>.</directory>
+        <directory>${module-path}</directory>
         <includes>
           <include>conf/**/*</include>
         </includes>
Index: dev-tools/maven/lucene/grouping/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/grouping/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/grouping/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Lucene Grouping Module</description>
   <properties>
     <module-directory>lucene/grouping</module-directory>
-    <build-directory>../build/grouping</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -66,11 +67,8 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
         <directory>${project.build.testSourceDirectory}</directory>
Index: dev-tools/maven/lucene/misc/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/misc/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/misc/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Miscellaneous Lucene extensions</description>
   <properties>
     <module-directory>lucene/misc</module-directory>
-    <build-directory>../build/misc</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -61,11 +62,8 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
         <directory>${project.build.testSourceDirectory}</directory>
Index: dev-tools/maven/lucene/spatial/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/spatial/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/spatial/pom.xml.template	(working copy)
@@ -35,7 +35,8 @@
   </description>
   <properties>
     <module-directory>lucene/spatial</module-directory>
-    <build-directory>../build/spatial</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <dependencies>
     <dependency> 
@@ -61,14 +62,11 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
-        <directory>src/test-files</directory>
+        <directory>${module-path}/src/test-files</directory>
       </testResource>
     </testResources>
   </build>
Index: dev-tools/maven/lucene/highlighter/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/highlighter/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/highlighter/pom.xml.template	(working copy)
@@ -35,7 +35,8 @@
   </description>
   <properties>
     <module-directory>lucene/highlighter</module-directory>
-    <build-directory>../build/highlighter</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -68,11 +69,8 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
         <directory>${project.build.testSourceDirectory}</directory>
Index: dev-tools/maven/lucene/sandbox/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/sandbox/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/sandbox/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Lucene Sandbox</description>
   <properties>
     <module-directory>lucene/sandbox</module-directory>
-    <build-directory>../build/sandbox</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -65,11 +66,8 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
         <directory>${project.build.testSourceDirectory}</directory>
Index: dev-tools/maven/lucene/join/pom.xml.template
===================================================================
--- dev-tools/maven/lucene/join/pom.xml.template	(revision 1334608)
+++ dev-tools/maven/lucene/join/pom.xml.template	(working copy)
@@ -33,7 +33,8 @@
   <description>Lucene Join Module</description>
   <properties>
     <module-directory>lucene/join</module-directory>
-    <build-directory>../build/join</build-directory>
+    <top-level>../../..</top-level>
+    <module-path>${top-level}/${module-directory}</module-path>
   </properties>
   <scm>
     <connection>
@@ -66,11 +67,8 @@
     </dependency>
   </dependencies>
   <build>
-    <directory>${build-directory}</directory>
-    <outputDirectory>${build-directory}/classes/java</outputDirectory>
-    <testOutputDirectory>${build-directory}/classes/test</testOutputDirectory>
-    <sourceDirectory>src/java</sourceDirectory>
-    <testSourceDirectory>src/test</testSourceDirectory>
+    <sourceDirectory>${module-path}/src/java</sourceDirectory>
+    <testSourceDirectory>${module-path}/src/test</testSourceDirectory>
     <testResources>
       <testResource>
         <directory>${project.build.testSourceDirectory}</directory>
Index: build.xml
===================================================================
--- build.xml	(revision 1334608)
+++ build.xml	(working copy)
@@ -83,9 +83,10 @@
   </target>
 
   <property name="version" value="4.0-SNAPSHOT"/>
+  <property name="maven-build-dir" value="maven-build"/>
   <target name="get-maven-poms"
-          description="Copy Maven POMs from dev-tools/maven/ to their target locations">
-    <copy todir="." overwrite="true">
+          description="Copy Maven POMs from dev-tools/maven/ to ${maven-build-dir}/">
+    <copy todir="${maven-build-dir}" overwrite="true">
       <fileset dir="${basedir}/dev-tools/maven"/>
       <filterset begintoken="@" endtoken="@">
         <filter token="version" value="${version}"/>
@@ -94,6 +95,11 @@
     </copy>
   </target>
 
+  <target name="clean-maven-build"
+          description="Remove ">
+    <delete failonerror="true" dir="${maven-build-dir}/"/>
+  </target>
+
   <target name="generate-maven-artifacts"
           description="Generate Maven Artifacts for Lucene and Solr">
     <property name="maven.dist.dir"  location="dist/maven" />
